### ISSUE TITLE
recent change losing custom shortcuts ?

### ISSUE THREAD

**Issue by phil20099** [darktable-org/darktable#11502]
**Describe the bug/issue**
With a recent upgrade, say in the last few days, I've lost some of my custom shortcuts.

I restored the 4 shortcut files in the ~/.config folder from yesterday.

Before running darktable doing  _grep 'apply BW' short*_ shows as below, which is correct.  They are 5 styles to apply a B&W filter via the colour calibration module.

shortcutsrc:b;shift=global/styles/apply BW|blue
shortcutsrc:g;shift=global/styles/apply BW|green
shortcutsrc:r;shift=global/styles/apply BW|red
shortcutsrc:t;shift=global/styles/apply BW|orange
shortcutsrc:y;shift=global/styles/apply BW|yellow
shortcutsrc.backup:b;shift=global/styles/apply BW|blue
shortcutsrc.backup:g;shift=global/styles/apply BW|green
shortcutsrc.backup:r;shift=global/styles/apply BW|red
shortcutsrc.backup:t;shift=global/styles/apply BW|orange
shortcutsrc.backup:y;shift=global/styles/apply BW|yellow
shortcutsrc.edit:b;shift=global/styles/apply BW|blue
shortcutsrc.edit:g;shift=global/styles/apply BW|green
shortcutsrc.edit:r;shift=global/styles/apply BW|red
shortcutsrc.edit:t;shift=global/styles/apply BW|orange
shortcutsrc.edit:y;shift=global/styles/apply BW|yellow

When I run darktable none of these shortcuts work e.g. shift + r to apply my black and white red filter.

Whilst darktable is running the grep shows as below, so no longer showing these in the shortcutsrc file :
shortcutsrc.backup:b;shift=global/styles/apply BW|blue
shortcutsrc.backup:g;shift=global/styles/apply BW|green
shortcutsrc.backup:r;shift=global/styles/apply BW|red
shortcutsrc.backup:t;shift=global/styles/apply BW|orange
shortcutsrc.backup:y;shift=global/styles/apply BW|yellow
shortcutsrc.edit:b;shift=global/styles/apply BW|blue
shortcutsrc.edit:g;shift=global/styles/apply BW|green
shortcutsrc.edit:r;shift=global/styles/apply BW|red
shortcutsrc.edit:t;shift=global/styles/apply BW|orange
shortcutsrc.edit:y;shift=global/styles/apply BW|yellow

The issue looks to be shortcuts linked to a style as my shortcuts to presets (e.g. tone equalizer / contrast tone curve soft) are still working.

Within darktable I then recreated the shortcut to shift + r.  I could create the shortcut however on applying it via shift + r it says the style has been applied but nothing happens.  It should add a module to the top of the history stack.

If I manually apply the style, either via the lighttable or the quick access icon bottom left of the darkroom, it works as expected.

**Expected behavior**
I expect the custom shortcuts to remain.

**Platform**
3.9.0~git1030.d5a8dd5d26-1 on Ubuntu 20.04

**Comment by phil20099** [comment]
I've built 3.9.0+990~ge6e5946a1 from 10th April and the shortcuts are working as expected i.e. they aren't being lost.

Creating a new shortcut to a style also worked.

**Comment by phil20099** [comment]
Thanks for the explanation.

I think the issue with me not being able to recreate the shortcuts is the style names contain a vertical bar.

The styles are named BW|red, BW|green etc so they nest under BW when looking at the styles in the darkroom.

- I renamed BW|red to BW-red, recreated the shortcut and it worked.
- I renamed BW-red back to BW|red, recreated the shortcut and it now works.

Without this renaming, in the styles section of the shortcuts the style names omit everything up to and including the vertical bar. In the example below I've already renamed the red and blue filters and I'm still do to green.

![1  styles](https://user-images.githubusercontent.com/60315272/163017763-19a978a2-3a15-4108-9067-16b6b9cf49e2.png)

This isn't an issue for me as I only have a few styles however I wanted to highlight just in case others experience the same issue.
